cirq-core/cirq/sim/state_vector.py

```diff
@@ -325,30 +326,24 @@ def measure_state_vector(
 
 def _probs(state: np.ndarray, indices: Sequence[int], qid_shape: Tuple[int, ...]) -> np.ndarray:
     """Returns the probabilities for a measurement on the given indices."""
-    tensor = np.reshape(state, qid_shape)
-    # Calculate the probabilities for measuring the particular results.
-    if len(indices) == len(qid_shape):
-        # We're measuring every qudit, so no need for fancy indexing
-        probs = np.abs(tensor) ** 2
-        probs = np.transpose(probs, indices)
-        probs = probs.reshape(-1)
+    state = state.reshape((-1,))
+    probs = np.abs(state) ** 2
+    not_measured = [i for i in range(len(qid_shape)) if i not in indices]
+    if linalg.can_numpy_support_shape(qid_shape):
+        # Use numpy transpose if we can since it's more efficient.
+        probs = probs.reshape(qid_shape)
+        probs = np.transpose(probs, list(indices) + not_measured)
```

Comment:
Looking at the usage of `np.transpose` elsewhere in the codebase, it looks like there is another place with similar usage: https://github.com/quantumlib/Cirq/blob/a95f009617bcdc613cfbca169fbedbfea1de85b7/cirq-core/cirq/sim/density_matrix_utils.py#L195-L213

Reply:
this is an exact copy of the code I'm modifying :disappointed: I refactored this part into a seprate method which I call from both locations